fake_geo_images/fakegeoimages.py

```diff
@@ -0,0 +1,200 @@
+            data_type: Rasterio datatype as string.
+            out_dir: Path where the image should be created (default ".").
+            coord_ref_sys: EPSG identifier of used coordinate reference system (default 3837).
+            nodata: Value representing nodata within each raster band, default is 0. If set to -1 no nodata value set.
+            nodata_fill: number of no data pixels to set in top left image (in x and y).
+            cog: output is a cloud-optimized geotiff. Only makes sense for larger images where block size matters.
+
+        Example:
+            ```python
+            input_img, an_array = FakeGeoImage(
+                10, 10, 4, "uint16", input_dir, nodata_fill=3
+            ).create(seed=45)
+            ```
+        """
+
+        self.xsize = xsize
+        self.ysize = ysize
+        self.num_bands = num_bands
+        self.data_type = data_type
+        self.out_dir = out_dir
+        self.crs = coord_ref_sys
+        self.nodata_fill = nodata_fill
+        if nodata == -1:
+            self.nodata = None
+        else:
+            self.nodata = nodata
+        self.cog = cog
+
+    def add_img_pattern(self, seed: Optional[int]) -> np.ndarray:
+        """
+        Simulate a five classes optical image.
+
+        Args:
+            seed: A random seed number. Ensures reproducibility.
+
+        Returns:
+            Numpy array representing simulated image.
+        """
+        if seed is not None:
+            np.random.seed(seed)
+
+        image, _ = skimage.draw.random_shapes(
+            (self.ysize, self.xsize),
+            max_shapes=50,
+            min_shapes=25,
+            multichannel=False,
+            allow_overlap=True,
+            random_seed=seed,
+        )
+        # Assign shape values to output classes
+        image[image < 55] = 1
+        image[(image >= 55) & (image < 105)] = 2
+        image[(image >= 105) & (image < 155)] = 3
+        image[(image >= 155) & (image < 205)] = 4
+        image[(image >= 205) & (image <= 255)] = 5
+
+        # Create bands having relevant values for all output classes
+        bands = []
+        band_idx = 0
+        while band_idx < self.num_bands:
+            data_ar = np.zeros_like(image, dtype=self.data_type)
+
+            for class_idx, lc_class in enumerate(LC_CLASSES.values(), 1):
+                # Add Gaussian noise
+                mask_ar = np.random.normal(
+                    lc_class["avg"][band_idx], lc_class["std"][band_idx], image.shape
+                )
+                data_ar[image == class_idx] = mask_ar[
+                    image == class_idx
+                ]  # pylint: disable=unsubscriptable-object
+            # Apply median filter to simulate spatial autocorrelation
+            data_ar = (signal.medfilt(data_ar)).astype(self.data_type)
+            data_ar = np.clip(data_ar, 1, None)
+            bands.append(data_ar)
+            band_idx += 1
+
+        return bands
+
+    def create(
```

Comment:
could be good to split this up in `create_array` and `create_file` to separate concerns. Can't imagine many usescases where I would need both returned & saves ressources.